draft-irtf-cfrg-opaque.md

```diff
@@ -558,46 +653,51 @@ Steps:
+The `EnvelopeMode` specifies the structure and encoding of the
+corresponding `InnerEnvelope`. This document specifies the values
+of the two aforementioned modes:
 
 ~~~
 enum {
- internal(1),
- external(2),
- (255)
+  internal(1),
+  external(2),
+  (255)
 } EnvelopeMode;
 ~~~
 
-Each `EnvelopeMode` defines its own `InnerEnvelope` structure and must implement the following interface:
-- `inner_env, client_public_key = BuildInnerEnvelope(random_pwd, nonce, client_private_key)`: build and return the
-  mode's `InnerEnvelope` structure and the client's public key.
-- `client_private_key, client_public_key = RecoverKeys(random_pwd, nonce, inner_env)`: recover and return the
-  client's private and public keys for the AKE protocol.
+Each `EnvelopeMode` defines its own `InnerEnvelope` structure and must implement
+the following interface:
 
-The implementations of this interface for both `internal` and `external` modes are in {{internal-mode}}
-and {{external-mode}}, respectively.
+- `inner_env, client_public_key = BuildInnerEnvelope(random_pwd, nonce, client_private_key)`:
+  build and return the mode's `InnerEnvelope` structure and the client's public key.
+- `client_private_key, client_public_key = RecoverKeys(random_pwd, nonce, inner_env)`:
+  recover and return the client's private and public keys for the AKE protocol.
 
-The size of the envelope may vary between modes. If applications implement {{preventing-client-enumeration}}, they
-MUST use the same envelope mode throughout their lifecycle in order to avoid activity leaks due to mode switching.
+The implementations of this interface for both `internal` and `external` modes
+are in {{internal-mode}} and {{external-mode}}, respectively.
+
+The size of the envelope may vary between modes. If applications implement
+{{preventing-client-enumeration}}, they MUST use the same envelope mode throughout
+their lifecycle in order to avoid activity leaks due to mode switching.
 
 ### Internal mode {#internal-mode}
 
-In this mode, the client's private and public keys are deterministically derived from the OPRF output.
+In this mode, the client's private and public keys are deterministically derived
+from the OPRF output.
 
-With the internal key mode the `EnvelopeMode` value MUST be `internal` and the `InnerEnvelope` is empty,
-and the size `Ne` of the serialized `Envelope` is Nn + Nm.
+With the internal key mode the `EnvelopeMode` value MUST be `internal` and the
+`InnerEnvelope` is empty, and the size `Ne` of the serialized `Envelope` is Nn + Nm.
 
-To generate the private key OPAQUE-3DH implements `DeriveAkeKeyPair(seed)` as follows:
+To generate the private key OPAQUE-3DH implements `DeriveAuthKeyPair(seed)` as follows:
 
 ~~~
-DeriveAkeKeyPair(seed)
+DeriveAuthKeyPair(seed)
 
 Parameter:
-- dst, domain separation tag for HashToScalar set to "OPAQUE-HashToScalar"
+- dst, domain separation tag for HashToScalar set to "OPAQUE-HashToScalar".
```

Comment:
Can we just omit this as a parameter, and hard-code "OPAQUE-HashToScalar" on line 707?

Reply:
So that line 707 reads:

1. private_key = HashToScalar(seed, dst = "OPAQUE-HashToScalar")